src/hammer-vlsi/hammer_vlsi/driver.py

```diff
@@ -448,6 +575,123 @@ def run_par(self, hook_actions: Optional[List[HammerToolHookAction]] = None, for
+        :return: drc.inputs.* settings generated from output_dict,
+                 or None if output_dict was invalid
+        """
+        try:
+            result = {
+                "drc.inputs.top_module": output_dict["par.inputs.top_module"],
+                "drc.inputs.layout_file": output_dict["par.outputs.output_gds"],
+                "vlsi.builtins.is_complete": False
+            }  # type: Dict[str, Any]
+            return result
+        except KeyError:
+            # KeyError means that the given dictionary is missing output keys.
+            return None
+
+    @staticmethod
+    def par_output_to_lvs_input(output_dict: dict) -> Optional[dict]:
+        """
+        Generate the appropriate inputs for running LVS from the
+        outputs of par run.
+        Does not merge the results with any project dictionaries.
+        :param output_dict: Dict containing par.outputs.*
+        :return: lvs.inputs.* settings generated from output_dict,
+                 or None if output_dict was invalid
+        """
+        try:
+            result = {
+                "lvs.inputs.top_module": output_dict["par.inputs.top_module"],
+                "lvs.inputs.layout_file": output_dict["par.outputs.output_gds"],
+                "lvs.inputs.schematic_files": [output_dict["par.outputs.output_netlist"]], #TODO john add ILM netlists
+                "lvs.inputs.power_nets": output_dict["par.outputs.power_nets"],
+                "lvs.inputs.ground_nets": output_dict["par.outputs.ground_nets"],
+                "lvs.inputs.hcells_list": output_dict["par.outputs.hcells_list"],
+                "vlsi.builtins.is_complete": False
+            }  # type: Dict[str, Any]
+            return result
+        except KeyError:
+            # KeyError means that the given dictionary is missing output keys.
+            return None
+
+    def run_drc(self, hook_actions: Optional[List[HammerToolHookAction]] = None, force_override: bool = False) -> Tuple[
+        bool, dict]:
+        """
+        Run DRC on a given database.
+        TODO fill me in
+        """
+        if self.drc_tool is None:
+            self.log.error("Must load DRC tool before calling run_drc")
+            return False, {}
+
+        self.log.info("Starting DRC check with tool '%s'" % (self.drc_tool.name))
+
+        if hook_actions is None:
+            hooks_to_use = self.post_custom_drc_tool_hooks
+        elif force_override:
+            hooks_to_use = hook_actions
+        else:
+            hooks_to_use = hook_actions + self.post_custom_drc_tool_hooks
+
+        run_succeeded = self.drc_tool.run(hooks_to_use)
+        if not run_succeeded:
+            self.log.error("DRC tool %s failed! Please check its output." % self.drc_tool.name)
+            # Allow the flow to keep running, just in case
+
+        # Record output from the drc_tool into the JSON output
+        # John: Do we need this??
+        output_config = deepdict(self.project_config)
```

Comment:
This should follow the current pattern (check synthesis/par tools) - this shouldn't include the base project config I think.

Reply:
Fixed.